File: task-03/sub-task 3/Elixir/code.ex
IO.puts "Enter an odd number of rows for the diamond:"
n = String.to_integer(IO.gets(""))

for i <- 0..div(n, 2) do
  IO.puts String.duplicate(" ", div(n, 2) - i) <> String.duplicate("*", 2 * i + 1)
end

for i <- div(n, 2) - 1..0 do
  IO.puts String.duplicate(" ", div(n, 2) - i) <> String.duplicate("*", 2 * i + 1)
end

File: task-03/sub-task 3/Haskell/code.hs
main = do
    putStrLn "Enter an odd number of rows for the diamond:"
    input <- getLine
    let n = read input :: Int

    -- Upper part including the middle row
    mapM_ putStrLn [replicate ((n `div` 2) - i) ' ' ++ replicate (2 * i + 1) '*' | i <- [0..(n `div` 2)]]

    -- Lower part (excluding the middle row)
    mapM_ putStrLn [replicate ((n `div` 2) - i) ' ' ++ replicate (2 * i + 1) '*' | i <- reverse [0..(n `div` 2 - 1)]]
